### PR TITLE
Changing Id from long to String and adding API

### LINKED ISSUE #48: Add APIs related to Users and Aggregator
**Describe the bug**
We need an API which connects studios to users, and users to skills and roles both at macroservice and aggregator level.

**Expected behavior**
Feign Clients should pick up the API connected to the aggregator

**Additional context**
- Make sure to have appropriate test cases(both unit and integration, also on the server) to confirm the working as well.

### PR DESCRIPTION
**Pull request by picksitquick**
Fixes #48 

## Proposed Changes

  - Added API related to get Users by Studios using Email
  - Updated the Id field from long to String

### CODE REVIEW

File: src/main/java/com/gamedoora/backend/userservices/assembler/UserServicesAssembler.java [hunk 1]
```diff
@@ -75,13 +75,13 @@ public UserDTO createUsers(UserDTO userDto) {
     return userDto;
   }
 
-  public UserDTO updateUsers(long id, UserDTO userDto) {
+  public UserDTO updateUsers(String emailId, UserDTO userDto) {
 
-    Optional<Users> usersRes = usersRepository.findById(id);
-    if (usersRes.isPresent()) {
+    Users usersRes = usersRepository.findByEmail(emailId);
+    if (usersRes.getEmail().isEmpty()) {
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/backend/userservices/assembler/UserServicesAssembler.java:81]: Can you please explain this check?

**Review by aprajshekhar — CHANGES_REQUESTED** [review]

File: src/main/java/com/gamedoora/backend/userservices/assembler/UserServicesAssembler.java (hunk 1 above)
Comment (picksitquick) [reply src/main/java/com/gamedoora/backend/userservices/assembler/UserServicesAssembler.java:81]: This is to make sure that if a user with the given email Id does not exists, return null, just like we previously handle the checks, only difference being this time is that we are not using optional tag so we need to verify whether there exists any such user or not, if not then return null.

Comment (picksitquick) [reply src/main/java/com/gamedoora/backend/userservices/assembler/UserServicesAssembler.java:81]: This needs to be modified as isEmpty() will check length and not for null.

File: src/main/java/com/gamedoora/backend/userservices/api/UserServicesController.java [hunk 2]
```diff
@@ -94,6 +94,13 @@ public ResponseEntity<UserDTO> getUserByEmail(@RequestParam(required = true) Str
     return createResponse(getUserServicesAssembler().getUserByEmail(email), HttpStatus.OK);
   }
 
+  @GetMapping(
+          value = "/studios/emailId",
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/backend/userservices/api/UserServicesController.java:98]: Email Id can have characters that can interfere with path. QueryParam would be a better option.

File: src/main/java/com/gamedoora/backend/userservices/api/UserServicesController.java [hunk 3]
```diff
@@ -36,21 +36,21 @@ public ResponseEntity<UserDTO> createUsers(@RequestBody UserDTO usersDto) {
   }
 
   @PutMapping(
-      value = "/{id}",
+      value = "/{emailId}",
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/backend/userservices/api/UserServicesController.java:39]: Email Id can have characters that can interfere with path. QueryParam would be a better option.

File: src/main/java/com/gamedoora/backend/userservices/api/UserServicesController.java [hunk 4]
```diff
@@ -36,21 +36,21 @@ public ResponseEntity<UserDTO> createUsers(@RequestBody UserDTO usersDto) {
...
-      throw new NotFoundException(MessageFormat.format("User by id {0} not found", id));
+      throw new NotFoundException(MessageFormat.format("User by id {0} not found", emailId));
     }
     return createResponse(userDTO, HttpStatus.OK);
   }
 
   @DeleteMapping("/{id}")
-  public ResponseEntity<HttpStatus> deleteUsers(@PathVariable("id") long id) {
-    getUserServicesAssembler().deleteUsers(id);
+  public ResponseEntity<HttpStatus> deleteUsers(@PathVariable("emailId") String emailId) {
```
Comment (aprajshekhar) [inline src/main/java/com/gamedoora/backend/userservices/api/UserServicesController.java:52]: Email Id can have characters that can interfere with path. QueryParam would be a better option. Also, the Path id is still `id`  in the annotation :)

**Review by aprajshekhar — CHANGES_REQUESTED** [review]
A suggestion regarding PathParam v/s QueryParam. Thanks!

**Review by aprajshekhar — APPROVED** [review]
Thanks for the changes. LGTM

**Pull request merged**